tests/accelerators/test_common.py

```diff
@@ -22,3 +27,50 @@ def test_auto_device_count(device_count_mock):
     assert GPUAccelerator.auto_device_count() == 2
     assert TPUAccelerator.auto_device_count() == 8
     assert IPUAccelerator.auto_device_count() == 4
+
+
+def test_pluggable_accelerator(tmpdir):
+    class TestAccelerator(Accelerator):
+        @staticmethod
+        def parse_devices(devices):
+            return devices
+
+        @staticmethod
+        def get_parallel_devices(devices):
+            return [torch.device("cpu")] * devices
+
+        @staticmethod
+        def auto_device_count():
+            return 1
+
+        @staticmethod
+        def is_available():
+            return True
+
+    model = BoringModel()
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        limit_train_batches=2,
+        limit_val_batches=2,
+        max_epochs=1,
+        accelerator=TestAccelerator(),
+        devices=2,
+        strategy="ddp",
+    )
+    trainer.fit(model)
```

Comment:
Can we avoid having to call `fit` twice in this test? There's no need to train for these assertions

Reply:
Good catch! Updated.